src/crimson/mon/MonClient.cc

```diff
@@ -1039,6 +1039,8 @@ seastar::future<> Client::reopen_session(int rank)
       return seastar::now();
     }
     return active_con->renew_rotating_keyring();
+  }).then([this] {
+    return sub.reload() ? renew_subs() : seastar::now();
```

Comment:
makes sense to me, the policy used for connecting to monitor from osd is lossy, so we are on our own. if we want to ensure the message is delivered to peer.

Reply:
Exactly.